test/test_spiner_transform.cpp

```diff
@@ -0,0 +1,95 @@
+#include <array>
+#include <cmath>
+#include <cstdio>
+#include <cstdlib>
+#include <iostream> // debug
+
+#include <ports-of-call/portability.hpp>
+#include <singularity-eos/eos/eos_spiner_common.hpp>
+#include <singularity-eos/eos/eos_spiner_sie_transforms.hpp>
+
+#include <spiner/databox.hpp>
+#include <spiner/interpolation.hpp>
+
+#ifndef CATCH_CONFIG_FAST_COMPILE
+#define CATCH_CONFIG_FAST_COMPILE
+#include <catch2/catch_test_macros.hpp>
+#endif
+
+#include <test/eos_unit_test_helpers.hpp>
+
+#ifdef SPINER_USE_HDF5
+#include <singularity-eos/base/spiner_table_utils.hpp>
+#endif
+
+#ifdef SPINER_USE_HDF5
+#ifdef SINGULARITY_TEST_SESAME
+
+using namespace singularity;
+using spiner_common::DataBox;
+using spiner_common::Grid_t;
+using spiner_common::to_log;
+
+PORTABLE_INLINE_FUNCTION bool isVeryClose(Real a, Real b, Real eps = 1e-14) {
+  return fabs(b - a) / (fabs(a + b) + 1e-20) <= eps;
+}
+
+struct TestDataContainer {
+  Real lRhoOffset = 0.5;
+
+  using RG = Spiner::RegularGrid1D<Real>;
+  Grid_t grid{{RG(1.0, 10.0, 100)}};
+  DataBox sieCold;
+
+  TestDataContainer() {
+    int N = static_cast<int>(grid.nPoints());
+    sieCold.resize(N);
+    sieCold.setRange(0, grid);
+    for (int i = 0; i < grid.nPoints(); ++i) {
+      Real lRho = grid.x(i);
+      Real e_cold = 3.0 * lRho + 5.0;
+      sieCold(i) = e_cold;
+    }
+  }
+};
+
+SCENARIO("ShiftTransform and NullTransform behave correctly", "[TransformTest]") {
+  TestDataContainer data;
+
+  ShiftTransform<TestDataContainer> shiftTransform(data);
+  NullTransform<> nullTransform;
+
+  GIVEN("A density with a known linear cold-curve in log-space") {
+    Real rho = 10.0;
+    Real lRho = to_log(rho, data.lRhoOffset);
+    Real cold_curve_value = 3.0 * lRho + 5.0;
```

Comment:
If you use my suggestion above, this would become
```suggestion
    Real cold_curve_value = data.e_cold_fun(lRho);
```